### ISSUE TITLE
Generated shim has an ugly icon

### ISSUE THREAD

**Issue by kodemeister** [aloneguid/win-shim#12]
1. Make a shim pointing to any app with icon, say Notepad:
```
shmake.exe -i notepad.exe -o sample.exe
```

2. Open Explorer and check the mirrored icon of the generated shim. It looks much worse compared to the original one.

![icons](https://github.com/aloneguid/win-shim/assets/1043953/a679bddc-b3c9-4b40-8171-9526012e375e)

I suspect this happens because **shmake doesn't copy icon groups from the input executable**. According to [docs](https://learn.microsoft.com/en-us/windows/win32/menurc/icons), Windows chooses the most appropriate icon to display based on icon groups but not plain icon resources.

Let's inspect the resources of Notepad app:

![notepad](https://github.com/aloneguid/win-shim/assets/1043953/5961f078-78e3-40fd-a513-6aa15ac06bc9)

As we see, Notepad has an icon group with ID 2 containing thirteen icons of various resolutions, including high quality ones.

Now the generated shim:

![sample](https://github.com/aloneguid/win-shim/assets/1043953/d8498731-8386-455c-a3b4-03d7db7a6db9)

There is an icon group with ID 101 containing only three low-res icons. That's why the generated shim appears ugly in Explorer.

Where did the icon group 101 come from? It is actually a leftover from the win-shim's default "umbrella" icon. Initially it contained three umbrella icons of resolutions 32x32, 64x64 and 256x256. During the shim creation umbrella icons were overwritten by Notepad icons. Now the icon group 101 points to the first three Notepad icons of much lower resolution. It is what causes the ugly appearance.

In summary, I believe shmake should copy icon groups from the input executable (numbered 2, 300, 301 and 302 in our particular case) and also remove umbrella icon group 101 to avoid unwanted leftovers and glitches.

**Comment by mnivet** [comment]
I’ve experienced the same issue. But since our need of shim.exe is on executables we compile ourself, for now we have just aligned the icons Embedded in our exe to match the icon groups from shim.exe.

I’ll be happy to see the fix to that issue officially integrated.
